### DISCUSSION TITLE
Docker extension on code server support

### DISCUSSION THREAD

**Discussion by ndutg** [coder/code-server#6915] [General] [answered]
Hi guys,

I am using code-server in a docker environment and I am trying to get the docker extension working. I installed it from the side tab but it keeps saying 'Failed to connect. Is Docker installed?' I have mounted the docker.sock in my volume in the YAML file. I am not sure what I am doing wrong or if this extension even works with code-server. Any help or suggestions on this will be appreciated. Cheers.

**Comment by robflate** [comment, marked as answer]
If you've already mounted the docker sock, you just need to exec into the code-server container;

`docker exec -it code-server /bin/bash`

and install the Docker CLI;

```
apt-get update
apt-get install -y docker.io
```

then install the Docker extension and it will work.

Note: The Docker CLI will not persist between container restarts. For that you could roll your own or use the [LinuxServer.io image](https://hub.docker.com/r/codercom/code-server) with the `DOCKER_MODS: "linuxserver/mods:universal-docker"` env var set.

**Reply by ndutg** [reply]
Thanks man, just the perfect response. This worked for me :)
